src/universal/README.html

```diff
@@ -0,0 +1,69 @@
+<h2 id="packages">Packages</h2>
+<p>A <em>package</em> is a directory of files. It will be uploaded to the output location. Files within the package may be used by <code>run</code> commands by specifying a <em>file</em> resource. The most common use is for directories containing script files.</p>
+<pre><code>package {id:&lt;package-id&gt;, source:&lt;source-url&gt;}</code></pre>
+<p>where <em>package-id</em> is a unique name for the package and <em>source-url</em> is a file path to a local directory or <code>s3:</code> url pointing to a zip file (which will be downloaded and expanded).</p>
+<h2 id="run-command">Run Command</h2>
+<p>The <code>run</code> command indicates a system process to execute.</p>
+<pre><code>run &lt;args&gt;</code></pre>
+<p>where each argument is either a string (as defined above) or a <em>resource</em>. A <em>resource</em> is specified by <code>key:value</code> pairs separated by commas within curly braces. The different kinds of resources are:</p>
+<h4 id="inputsource-url-typefiledir"><code>{input:&lt;source-url&gt;, type:&lt;file|dir&gt;}</code></h4>
+<p>An input file or directory, where <em>source-url</em> is a local file path or <code>s3:</code> URL. At runtime, this will resolve to an absolute filesystem path. If not specified, the default type is <code>file</code>.</p>
+<h4 id="filefile-name-packagepackage-id"><code>{file:&lt;file-name&gt;,  package:&lt;package-id&gt;}</code></h4>
+<p>An individual file within a <em>package</em> declared earlier</p>
+<h4 id="outputoutput-id-typefiledir-suffixsuffix"><code>{output:&lt;output-id&gt;, type:&lt;file|dir&gt;, suffix:&lt;suffix&gt;</code>}</h4>
+<p>An output file or directory. At runtime, this will resolve to an absolute filesystem path. <strong>The system process must create a file/directory at that location when it finishes</strong>. If not specified, the default type is <code>file</code>. The output will be uploaded with a name of the form <code>&lt;output-id&gt;.&lt;hash-code&gt;&lt;suffix&gt;</code>.</p>
+<h4 id="refoutput-id"><code>{ref:&lt;output-id&gt;}</code></h4>
+<p>The output of a previous <code>run</code> command, matched by <em>output-id</em>. At runtime, this will resolve to an absolute local filesystem path.</p>
+<h1 id="example">Example</h1>
+<p>Here is an example of a prototypical Python-based workflow</p>
+<pre><code>set {imageDir: /home/ps-user/data/images, scriptDir: /home/ps-user/python/classification-scripts}
+
+package {id:scripts, source: $scriptDir}
+
+run python {file:TrainTestSplit.py, package: scripts} 
+    -input $imageDir 
+    -outTrain {output:trainImages, type:dir} 
+    -outTest {output:testImages, type:dir}
+    
+run python {file:TrainModel.py, package: scripts}
+    -images {ref: trainImages}
+    -output {output:modelFile}
+    
+ run python {file: EvalModel.py, package: scripts}
+     -images {ref:testImages}
+     -model {ref:modelFile}
+     -output {output: evaluation, suffix:&quot;.txt&quot;}</code></pre>
```

Comment:
This is an HTML version of the `PipeScript.md` file. I made it for the
binary release. If you do `sbt universal:packageBin` you get a zip file
with bin and lib directories and this README.